src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -412,6 +457,44 @@ std::shared_ptr<Pane> Pane::NavigateDirection(const std::shared_ptr<Pane> source
         return PreviousPane(sourcePane);
     }
 
+    // Check if moving up or down the tree
+    if (direction == FocusDirection::Parent)
+    {
+        if (const auto parent = _FindParentOfPane(sourcePane))
+        {
+            // Capture a weak reference to the source pane if it is a leaf so that
+            // we can keep the same terminal focused and take the same path back down
+            // if a child movement is requested.
+            parent->_previouslyFocusedTerminal = sourcePane->_IsLeaf() ? sourcePane->weak_from_this() : sourcePane->_previouslyFocusedTerminal;
+
+            return parent;
+        }
+        return nullptr;
+    }
+
+    if (direction == FocusDirection::Child)
+    {
+        if (!sourcePane->_IsLeaf())
+        {
+            auto child = sourcePane->_firstChild;
+            // If we've recorded an originally focused terminal try to find it
+            if (const auto prevFocus = sourcePane->_previouslyFocusedTerminal.lock())
+            {
+                // We default to the first child, so just check if the second child
+                // has the terminal we are looking for.
+                if (const auto p = sourcePane->_secondChild->FindPane(prevFocus->_id.value()))

```

Comment:
oh interesting. Okay so `_previouslyFocusedTerminal` will always be a leaf, but this logic here will try and find where `_previouslyFocusedTerminal` is from the two children. Would it be easier to just stash `sourcePane` always up on L468? I suppose the pane could move, so we'd still be heading towards it, wherever it is now. 

Okay I've convinced myself this is right

Reply:
I chose to store the leaf pane, and deliberately named this variable `*Terminal`, because the whole reason it is there is so that `GetLastFocusedTerminalControl` has an easy way to get the `TermControl` on the original leaf pane. Storing the path to the leaf makes the `child` movement a bit easier, but complicates `GetLastFocusedTerminalControl` as it then needs to follow the path instead of getting the control directly.

Reply:
Ended up having to store the path anyways to fix the issue where a terminal 'exit'ing would not set focus correctly when a parent was focused.